### ISSUE TITLE
can't get plugin rules by `CLIEngine#getRules`

### ISSUE THREAD

**Issue by fisker** [eslint/eslint#11871]
**Tell us about your environment**

* **ESLint Version:**
6.0.0

`cli.getRules().get(ruleId)` used can get plugin rule, now not.

Is this by design or a bug?

```js
new (require('eslint').CLIEngine)().getRules().has('yoda')
// -> true
```

```js
new (require('eslint').CLIEngine)({plugins: ['unicorn']}).getRules().has('unicorn/no-process-exit')
// -> false, should be true
```

**Comment by fisker** [comment]
After a few tries, I found after calling `executeOnText` or `executeOnFiles`, rules appeared, but calling `getConfigForFile` (which is I used before and works in  https://github.com/eslint/eslint/issues/10838 ) plugin rules are not loaded.

**Comment by mysticatea** [comment]
Thank you for your report.

Sounds like a bug.

**Comment by mysticatea** [comment]
Yes, currently, it returns core rules and the plugins rules that the last `execute*` method call used.
